### PR TITLE
Move persistence pointcuts to parent listener

### PR DESCRIPTION
**Pull request by taomoh**
Refactor to move JPA point-cuts to parent listener: `KafkaEntityListener`.
This will centralise persistence lifecycle processing in one listener and avoid code duplication. 
This will also eliminate the need to implement entity listeners as a child classes. It will suffice to declare `KafkaEntityListener` as entity listener for all entities.
For this to work, each entity class will now need to implement `Messageable` interface, also introduced in this change, to allow the message key to be specified

### CODE REVIEW

File: kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java [hunk 1]
```diff
@@ -1,32 +1,37 @@
...
 
   private final KafkaProducerService<T> kafkaProducerService;
 
   protected KafkaEntityListener(KafkaProducerService<T> kafkaProducerService) {
     this.kafkaProducerService = kafkaProducerService;
   }
 
-  public abstract String resolveMessageKey(T resource);
-
+  @PostPersist
```
Comment (JWeston-HO) [inline kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java:19]: Do we want to keep this as post and not change it to pre?

Comment (taomoh) [reply kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java:19]: @Pre works in tandem with the Transaction Management synchronisation. So we should change it to the @Pre annotation only once the transaction management synchronisation piece is added to kafka-commons

Comment (JWeston-HO) [reply kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java:19]: That makes sense to me.

**Review by JWeston-HO — APPROVED** [review]

File: kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
   protected void sendKafkaMessageOnCreate(T resource) {
```
Comment (daleelR) [inline kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java:20]: Do we need to re-consider the access modifier here and on the others?
Now that we don't need to directly call any methods as they are listener annotated could it go back to private or even public based on some of our previous discussion?

Comment (taomoh) [reply kafka-commons/src/main/java/uk/gov/homeoffice/digital/sas/kafka/listener/KafkaEntityListener.java]: As discussed, I made the methods private, constructor public and class final

**Review by daleelR — APPROVED** [review]

**Review by JWeston-HO — APPROVED** [review]

**Pull request merged**